### ISSUE TITLE
Int4(llama-2-chat-7b) converted model generates response in German language

### ISSUE THREAD

**Issue by bhardwaj-nakul** [openvinotoolkit/openvino_notebooks#1683]
**Describe the bug**

1. Install all the pip dependencies for latest 254-llmchatbot notebook
2.  Follow the steps to convert "llama-2-chat-7b" model to int4 format with default configuration.
3. Select device as "CPU"
4. Select model to run "INT4"
5. Run step to Load and compile the model
6. Set `max_new_token=500` and run `ov_model.generate` with prompt "Describe Intel in 100 words or less"

 **Expected behavior**

Output should be produced in English. However we are getting output in German language. 

1. Is there any issue with converting llama-2-chat-7b model into int4 format with OpenVino ?
2. Is the issue with latest openvino==2023.3.0 or nncf==2.9.0.dev0+84b46f58 ?

**Screenshots**
Screenshot 1
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/84007248/dc0277fa-1cee-4844-bd4c-2fb665c98b70)

Screenshot 2
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/84007248/db9fecb2-05e3-49b3-b75d-7074751e6834)

Screenshot 3
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/84007248/357021b4-a770-43ea-8084-533aa4eb82eb)

 **Installation instructions (Please mark the checkbox)**
- [x] I followed the installation guide at https://github.com/openvinotoolkit/openvino_notebooks#-installation-guide to install the notebooks. 

 
**Additional context**
I tried playing around with `model_compression_params` but it didn't help to resolve this issue.
```
"llama-2-chat-7b": {
    "mode": nncf.CompressWeightsMode.INT4_SYM,
    "group_size": 128,
    "ratio": 0.8,
},
```
```
"llama-2-chat-7b": {
    "mode": nncf.CompressWeightsMode.INT4_ASYM,
    "group_size": 128,
    "ratio": 0.8,
},
```
```
"llama-2-chat-7b": {
    "mode": nncf.CompressWeightsMode.INT4_SYM,
    "group_size": 64,
    "ratio": 0.8,
},
```
```
"llama-2-chat-7b": {
    "mode": nncf.CompressWeightsMode.INT4_SYM,
    "group_size": 64,
    "ratio": 0.6,
},
```

**Comment by OpenVINO-dev-contest** [comment]
Hi @bhardwaj-nakul Since this model is for propose of chat, could you help to try adding a system prompt before the question, and see if the issue happen again?

**Comment by AlexKoff88** [comment]
@andreyanufr, can you please take a look and provide an advice on what kind of weight compression configuration should lead to accurate results?

**Comment by andreyanufr** [comment]
@bhardwaj-nakul could you please try to run data aware compression:

`from functools import partial
import numpy as np
from nncf import compress_weights, Dataset
from nncf.parameters import CompressWeightsMode, SensitivityMetric
from transformers import AutoTokenizer
from transformers import AutoConfig
from compress_qwen_awq import OVQwenModel
from optimum.intel.openvino import OVModelForCausalLM
from datasets import load_dataset




def transform_func(item, tokenizer, input_shapes):
    text = item['text']

    tokens = tokenizer(text)

    res = {'input_ids': np.expand_dims(np.array(tokens['input_ids']), 0),
           'attention_mask': np.expand_dims(np.array(tokens['attention_mask']), 0)}

    if 'position_ids' in input_shapes:
        position_ids = np.cumsum(res['attention_mask'], axis=1) - 1
        position_ids[res['attention_mask'] == 0] = 1
        res['position_ids'] = position_ids

    for name, shape in input_shapes.items():
        if name in res:
            continue
        res[name] = np.zeros(shape)

    return res


def get_input_shapes(model, batch_size = 1):
    inputs = {}

    for val in model.model.inputs:
        name = val.any_name
        shape = list(val.partial_shape.get_min_shape())
        shape[0] = batch_size
        inputs[name] = shape

    return inputs

def compress(model, tokenizer, mode = CompressWeightsMode.INT4_SYM,
             group_size=128, ratio = 0.8,
             sm = SensitivityMetric.MAX_ACTIVATION_VARIANCE):
    dataset = load_dataset('wikitext', 'wikitext-2-v1', split='train')
    dataset = dataset.filter(lambda example: len(example["text"]) > 80)


    input_shapes = get_input_shapes(model)
    nncf_dataset = Dataset(dataset, partial(transform_func, tokenizer=tokenizer, input_shapes=input_shapes))

    model.model = compress_weights(model.model, mode=mode,
                                   group_size=group_size,
                                   ratio=ratio,
                                   dataset=nncf_dataset,
                                   sensitivity_metric=sm)

    return model`

**Comment by andrei-kochin** [comment]
@bhardwaj-nakul could you please try the example suggested?

**Comment by bhardwaj-nakul** [comment]
I tried the provided script. I'm getting below error

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/84007248/76d5f283-89c7-4e7e-90f0-d3f3bb710430)

The input model to this script is supposed to be FP16 IR model right ? I tried with this itself.
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/84007248/ca4b2b7e-66b6-4ea1-8475-c42e7c9b052c)
